liquid-compiler/src/grammar.pest

```diff
@@ -0,0 +1,56 @@
+WHITESPACE = _{" "}
+LINE_OR_SPACE = _{ NEWLINE | WHITESPACE }
+LiquidFile = ${ SOI ~ Element* ~ EOI }
+
+// Element-level parsing
+Element = _{ Expression | Tag | Raw } 
+
+TagStart = _{ (LINE_OR_SPACE* ~ "{%-") | "{%" }
+TagEnd =   _{ ("-%}" ~ LINE_OR_SPACE*) | "%}" }
+ExpressionStart = _{ (LINE_OR_SPACE* ~ "{{-") | "{{" }
+ExpressionEnd =   _{ ("-}}" ~ LINE_OR_SPACE*) | "}}" }
+
+Tag = !{ TagStart ~ Identifier ~ TagToken* ~ TagEnd }
+Expression = !{ ExpressionStart ~ FilterChain ~ ExpressionEnd }
+// Not allowing Tag/Expression Start/End might become a problem 
+// for {% raw %} tag: unclosed delimiters won't be accepted
+Raw = @{ (!(TagStart | ExpressionStart | TagEnd | ExpressionEnd) ~ ANY)+ }              
+
+
+// Inner parsing        
+Identifier = @{ (ASCII_ALPHA | "_")+ }
+
+Variable = { Identifier 
+           ~ ( ("." ~ Identifier)
+             | ("[" ~ Value ~ "]")
+             )* 
+           }
+Value = { Literal | Variable }
+Filter = { Identifier ~ (":" ~ Value ~ ("," ~ Value)*)? } 
```

Comment:
This is where we'd extend the grammar for #92 

Depending on architecture, we *might* be able to share named arguments with `for` blocks and others.